vortex-scalar/src/binary.rs

```diff
@@ -40,7 +42,7 @@ impl Scalar {
     pub fn binary(buffer: ByteBuffer, nullability: Nullability) -> Self {
```

Comment:
For constructor functions like this I usually prefer to take the `Arc<ByteBuffer>`, just in case the user already has one and can avoid deref/re-alloc

Reply:
But it's kinda gross I guess...

Reply:
we can also make the scalar fields public, which will give another way to initialize one if the user already has an `Arc`

Reply:
I think `fn binary(buffer: impl Into<Arc<ByteBuffer>>, nullability: Nullability)` is a good solution